src/components/ha-absolute-time.ts

```diff
@@ -0,0 +1,76 @@
+import { isSameDay } from "date-fns";
+import { PropertyValues, ReactiveElement } from "lit";
+import { customElement, property } from "lit/decorators";
+import { absoluteTime } from "../common/datetime/absolute_time";
+import type { HomeAssistant } from "../types";
+
+@customElement("ha-absolute-time")
+class HaAbsoluteTime extends ReactiveElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ attribute: false }) public datetime?: string | Date;
+
+  private _lastUpdate?: Date;
+
+  private _interval?: number;
+
+  public disconnectedCallback(): void {
+    super.disconnectedCallback();
+    this._clearInterval();
+  }
+
+  public connectedCallback(): void {
+    super.connectedCallback();
+    if (this.datetime) {
+      this._startInterval();
+    }
+  }
+
+  protected createRenderRoot() {
+    return this;
+  }
+
+  protected firstUpdated(changedProps: PropertyValues) {
+    super.firstUpdated(changedProps);
+    this._updateAbsolute();
+  }
+
+  protected update(changedProps: PropertyValues) {
+    super.update(changedProps);
+    this._updateAbsolute();
+  }
+
+  private _clearInterval(): void {
+    if (this._interval) {
+      window.clearInterval(this._interval);
+      this._interval = undefined;
+    }
+  }
+
+  private _startInterval(): void {
+    this._clearInterval();
+
+    // update every 60 seconds
```

Comment:
Why do we need to update an absolute time every 60 secs?

Reply:
We should update it every hour (HH:00)

Reply:
The format change if it's the same day or not.

```
 const _to = to ?? new Date();

  if (isSameDay(from, _to)) {
    return formatTime(from, locale);
  }
  if (isSameYear(from, _to)) {
    return formatShortDateTime(from, locale);
  }
  return formatShortDateTimeWithYear(from, locale);
  ```
  
The check is now done every hour.
At midnight, the display value should change from  8:23 AM` to `Aug 9, 2021, 8:23 AM`.

Reply:
Then we could just calc the ms to next day instead of doing it every hour? And if it is already not the same day we dont have to do it at all?

Reply:
That is a pretty big edge case, how many times will a page be open for so long without refresh/update, etc?